watch.go

```diff
@@ -41,11 +47,21 @@ loop:
 		select {
 		case <-watcher.Events:
 			for _, a := range args {
```

Comment:
could we just loop through taskCancellers directly? If it's set in a non-concurrent manner, we don't need as many checks as here.